static/panes/compiler.js

```diff
@@ -94,12 +94,41 @@ function Compiler(hub, container, state) {
     this.wantOptInfo = state.wantOptInfo;
     this.decorations = {};
     this.prevDecorations = [];
+    this.labels = {};
+    this.ctrlIsPressed = false;
     this.alertSystem = new Alert();
     this.alertSystem.prefixMessage = "Compiler #" + this.id + ": ";
 
     this.linkedFadeTimeoutId = -1;
     this.toolsMenu = null;
 
+    var labelUsageInstructions = [
+        "call",
```

Comment:
This is very x86-specific. Of course, a lot of Compiler Explorer is (unfortuantely) but it might be nice to support more here. I _think_ there's a way to support non-x86 (and non-jump).

My thoughts are it's not just jump instructions that refer to labels. Data address references, directives etc can all refer to labels. My instinct is to try and find the _label_ itself and not the "left hand side" of a label-needing operation.

Additionally we already "know" things that appear to be labels on the server. It might be better to annotate the lines on the server and use that information in the client, instead of duplicating the logic in two places.

I realise that's a bigger change, but it keeps the code and the parser in one place. And all the UI code here should remain pretty much as-is; just instead of finding the labels here, we just use the previously-annotated label information.

If you wanted to take a look, `lib/asm-parser.js` has all the horrible regexes and parser tidbits you could ever want (and more...), plus support for many architectures and a boatload of tests.